### PR TITLE
Master simplify sales team config qmo

### PR DESCRIPTION
**Pull request by tde-banana-odoo**
Reason:
  *Sales team configuration is too complex with fields you
   don't understand and that bring no added value
  *Remove the team type, every team should be able to handle
   a POS or a website for example
Contains:
  *Removal of the team_type, use_quotations, use_leads fields
  *Removal of the graph configuration fields
  *Adapting the filters that were based on the team_type
  *Clear distinction between the team dashboard behavior in crm
   from the one in sales
  *Cleansing of now dead code due to the removal of the previously
   mentionned fields

### CODE REVIEW

**Comment by qmo-odoo** [comment]
Task: https://www.odoo.com/web#id=1830105&action=333&active_id=965&model=project.task&view_type=form&menu_id=4720

Purpose : 
Simplify sales team config by removing dashboard configuration and team_type, use_quotations, use_leads fields. Anyy salesteam should be usable in any channel, allowing to group several channels into one team.

Behavior before PR: 
Users could configure the team dashboard graphs, the type of the team or wether or not the team should use leads or/and quotations

Behavior after PR:
No more dashboard graph configuration, teams who manage a pipeline will have a graph based on "crm.lead" which will display the number of opportunities/leads per week for the last month. Teams who don't manage a pipeline will have a graph based on sale.report which will display the total of SO (+POS Order if the team is in charge of a point of sale)

File: addons/sale_crm/models/crm_team.py [hunk 1]
```diff
@@ -1,12 +1,53 @@
...
+            teams_no_opp.update({'dashboard_button_name': _("Sales Analysis")})
+
+    def action_primary_channel_button(self):
+        action = self.env.ref('crm.crm_case_form_view_salesteams_opportunity').read()[0] 
+        if self._context.get('in_sales_app') or not self.use_opportunities:
+            action = self.env.ref('sale.action_order_report_so_salesteam').read()[0]
+        return action
+    
+    def _graph_get_model(self):
+        graph_model = 'sale.report'
```
Comment (jem-odoo) [inline addons/sale_crm/models/crm_team.py]: For all this methods (_graph_get_model, _graph_date_column, _graph_y_query, _graph_y_query, _graph_title_and_key), you definitely need to call super.
Place your condition before calling super, so you can take the priority and return the value the module provides, otherwise, return `super()`.
```
if "check if the current team and context are related to my module business":
     return "the value implemented in the module"
return super()  # this is not my business, so I let other modules handle this case
```

File: addons/sales_team/models/crm_team.py [hunk 2]
```diff
@@ -219,54 +179,27 @@ def get_week_name(start_date, locale):
...
+
+        for data_item in graph_data:
+            index = int((data_item.get('x_value') - start_date.isocalendar()[1]) % weeks_in_start_year)
+            values[index][y_field] = data_item.get('y_value')
 
         [graph_title, graph_key] = self._graph_title_and_key()
         color = '#875A7B' if '+e' in version else '#7c7bad'
         return [{'values': values, 'area': True, 'title': graph_title, 'key': graph_key, 'color': color}]
 
+
```
Comment (jem-odoo) [inline addons/sales_team/models/crm_team.py]: useless diff

File: addons/sales_team/models/res_users.py [hunk 3]
```diff
@@ -16,7 +16,7 @@ def create(self, vals):
         # Assign the new user in the sales team if there's only one sales team of type `Sales`
         user = super(ResUsers, self).create(vals)
         if user.has_group('sales_team.group_sale_salesman') and not user.sale_team_id:
-            teams = self.env['crm.team'].search([('team_type', '=', 'sales')])
+            teams = self.env['crm.team'].search([(1, '=', 1)])
```
Comment (jem-odoo) [inline addons/sales_team/models/res_users.py]: This will cost a search at each user creation, and it will only be usefull in the case we got only one team in the database. Isn't there a domain we can put here ?
Or, ask your PO if it is still required to put the user in a team at its creation.

File: addons/sales_team/views/crm_team_views.xml [hunk 4]
```diff
@@ -22,12 +20,29 @@
 
         <!-- Case Teams Action -->
         <record id="crm_team_salesteams_act" model="ir.actions.act_window">
+            <field name="name">Sales Teams</field>
+            <field name="res_model">crm.team</field>
+            <field name="view_type">form</field>
+            <field name="view_mode">kanban,form</field>
+            <field name="view_id" ref="crm_team_salesteams_search"/>
+            <field name="context">{'button_name': 'Sales Orders'}</field>
```
Comment (jem-odoo) [inline addons/sales_team/views/crm_team_views.xml]: I did not find where your are using this context key.
Moreover, this is not translatable. So, this is not a good idea to do this. You'd rather pass a technical context key, that will trigger (maybe in the context field handling the label name) the label itself and its translation.

File: addons/website_crm/models/crm_lead.py [hunk 5]
```diff
@@ -22,10 +22,10 @@ class Website(models.Model):
     _inherit = 'website'
 
     def _get_crm_default_team_domain(self):
-        if self.env.user.has_group('crm.group_use_lead'):
-            return [('use_leads', '=', True)]
-        else:
+        if not self.env.user.has_group('crm.group_use_lead'):
             return [('use_opportunities', '=', True)]
+        else:
+            return [(1, '=', 1)]
```
Comment (jem-odoo) [inline addons/website_crm/models/crm_lead.py]: You can just return empty array, and remove the `else` (you won a line)

File: addons/crm/views/crm_team_views.xml [hunk 6]
```diff
@@ -72,9 +89,9 @@
        <record id="action_report_crm_opportunity_salesteam" model="ir.actions.act_window">
             <field name="name">Pipeline Analysis</field>
             <field name="res_model">crm.lead</field>
-            <field name="view_mode">graph,pivot,tree,form</field>
-            <field name="domain">[('type', '=', 'opportunity'), ('team_id', '=', active_id)]</field>
-            <field name="context">{'active_test': True}</field>
+            <field name="view_mode">dashboard,graph,pivot,tree,form</field>
```
Comment (jem-odoo) [inline addons/crm/views/crm_team_views.xml]: no dashboard view in community

File: addons/pos_sale/views/pos_config_views.xml [hunk 7]
```diff
@@ -6,16 +6,20 @@
...
+                            <div class="text-muted">
+                                Sales are reported to the following sales team
+                            </div>
+                            <field name="crm_team_id"/>
+                        </div>
                     </div>
-                  </div>
                 </div>
-            </div>
+            </div>
```
Comment (tde-banana-odoo) [inline addons/pos_sale/views/pos_config_views.xml]: HumanLinter: tab added by mistake ?

File: addons/sales_team/models/crm_team.py [hunk 8]
```diff
@@ -22,24 +22,26 @@ class CrmTeam(models.Model):
...
     def _get_default_team_id(self, user_id=None):
-        if not user_id:
-            user_id = self.env.uid
-        company_id = self.sudo(user_id).env.user.company_id.id
-        team_id = self.env['crm.team'].sudo().search([
-            '|', ('user_id', '=', user_id), ('member_ids', '=', user_id),
-            '|', ('company_id', '=', False), ('company_id', 'child_of', [company_id])
-        ], limit=1)
-        if not team_id and 'default_team_id' in self.env.context:
+        if 'default_team_id' in self.env.context:
```
Comment (tde-banana-odoo) [inline addons/sales_team/models/crm_team.py]: This method is highly controversial and I would like not changing it. Why did you change its behavior ?

According to spec you have some default teams (pos / website) -> those should be trigerred only if the current user is not member or manager of a team. Own team > other default values.

File: addons/sales_team/models/crm_team.py [hunk 9]
```diff
@@ -298,6 +221,18 @@ def write(self, values):
             self._add_members_to_favorites()
         return res
 
+    @api.multi
+    def unlink(self):
```
Comment (tde-banana-odoo) [inline addons/sales_team/models/crm_team.py]: I would remove this override. Some people are working on a way to mark some data as master data. Anyway Odoo should still work without those teams data even if it means that some default values are not computed.

Comment (tde-banana-odoo) [reply addons/sales_team/models/crm_team.py]: After discussion, we keep it  because if you remove some data and then install website sale it crashes. We keep this code until a better solution to manage sales team is fonhd.

**Comment by tde-banana-odoo** [comment]
@robodoo r+
